### PR TITLE
test: fix test API_CurlImportPost_spec.js

### PR DESCRIPTION
**Pull request by Parthvi12**
## Description

Fixes test by removing third party api to TED api

#### How Has This Been Tested?
locally

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 1]
```diff
@@ -32,14 +32,14 @@ describe("Test curl import flow", function () {
...
-  it.skip("2. Test curl import flow for POST action with multipart form data", function () {
+  it("2. Test curl import flow for POST action with multipart form data", function () {
     localStorage.setItem("ApiPaneV2", "ApiPaneV2");
     cy.NavigateToApiEditor();
     _.dataSources.NavigateToDSCreateNew();
 
     cy.get(ApiEditor.curlImage).click({ force: true });
     cy.get("textarea").type(
-      "curl --request POST http://httpbin.org/post -F 'randomKey=randomValue' --form 'randomKey2=\"randomValue2\"'",
+      "curl --request POST http://host.docker.internal:5001/v1/mock-api/echo-multipart -F 'randomKey=randomValue' --form 'randomKey2=\"randomValue2\"'",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:42]: No hardcoding. Please have it as variable in dataSouces.json & bring it in here.

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 2]
```diff
@@ -32,14 +32,14 @@ describe("Test curl import flow", function () {
     });
   });
 
-  it.skip("2. Test curl import flow for POST action with multipart form data", function () {
+  it("2. Test curl import flow for POST action with multipart form data", function () {
     localStorage.setItem("ApiPaneV2", "ApiPaneV2");
     cy.NavigateToApiEditor();
     _.dataSources.NavigateToDSCreateNew();
 
     cy.get(ApiEditor.curlImage).click({ force: true });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:40]: Can we use ImportCurl method from Datasouces.ts helper?

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 3]
```diff
@@ -49,17 +49,10 @@ describe("Test curl import flow", function () {
...
-      cy.log(response.response.body);
-      cy.expect(response.response.body.responseMeta.success).to.eq(true);
-      // Asserting if the form key value are returned in the response
-      cy.expect(response.response.body.data.body.form.randomKey).to.eq(
-        "randomValue",
-      );
-      // Asserting the content type header set in curl import is multipart/form-data
-      cy.expect(
-        response.response.body.data.body.headers["Content-Type"],
-      ).contains("multipart/form-data;boundary");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:62]: This validation is important.. why is it removed?

**Review by Aishwarya-U-R — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js (hunk 3 above)
Comment (Parthvi12) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js]: I have added another method to validate the headers, hence removed

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 4]
```diff
@@ -32,34 +27,15 @@ describe("Test curl import flow", function () {
...
-    cy.importCurl();
-    cy.RunAPI();
-    cy.ResponseStatusCheck("200 OK");
-    cy.log("Ran the API successfully");
-    cy.get("@postExecute").then((response) => {
-      cy.log(response.response.body);
-      cy.expect(response.response.body.responseMeta.success).to.eq(true);
-      // Asserting if the form key value are returned in the response
-      cy.expect(response.response.body.data.body.form.randomKey).to.eq(
-        "randomValue",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:57]: How about this validation? Is this covered?

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js (hunk 3 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js]: Got it about the headers. Good job there to use the TS helper!

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js (hunk 4 above)
Comment (Parthvi12) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:57]: Good catch, added it back.

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 5]
```diff
@@ -1,25 +1,20 @@
 const apiwidget = require("../../../../locators/apiWidgetslocator.json");
 import ApiEditor from "../../../../locators/ApiEditor";
-import { dataSources } from "../../../../support/Objects/ObjectsCore";
+import * as _ from "../../../../support/Objects/ObjectsCore";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:3]: Import only needed objects from ObjectsCore. No `import *`

File: app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js [hunk 6]
```diff
@@ -32,34 +27,23 @@ describe("Test curl import flow", function () {
...
       );
-      // Asserting the content type header set in curl import is multipart/form-data
-      cy.expect(
-        response.response.body.data.body.headers["Content-Type"],
-      ).contains("multipart/form-data;boundary");
+      cy.get("@postExecute").then((response) => {
+        cy.log(response);
+        cy.expect(response.response.body.responseMeta.success).to.eq(true);
+        // Asserting if the form key value are returned in the response
+        cy.expect(response.response.body.data.body.data.randomKey).to.eq(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:39]: Can we check if we can use `agHelper.AssertNetworkDataNestedProperty` here? Change it, run locally & let me know!

Comment (Parthvi12) [reply app/client/cypress/e2e/Regression/ServerSide/ApiTests/API_CurlPOSTImport_spec.js:39]: Okay

File: app/client/cypress.config.ts [hunk 7]
```diff
@@ -24,10 +24,10 @@ export default defineConfig({
     openMode: 0,
   },
   e2e: {
-    baseUrl: "https://dev.appsmith.com/",
+    baseUrl: "http://localhost/",
     env: {
-      USERNAME: "xxxx",
-      PASSWORD: "xxx",
+      USERNAME: "parthvi12@appsmith.com",
+      PASSWORD: "qwerty",
```
Comment (Aishwarya-U-R) [inline app/client/cypress.config.ts:30]: Revert this merge please!

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**